### PR TITLE
MGDOBR-481: add WebhookOidcClient for sendToBridge action

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-481](https://issues.redhat.com/browse/MGDOBR-481)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [x] Check if code and Github action workflows have been modified and if the modification is safe
- [x] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java [hunk 1]
```diff
@@ -56,10 +58,11 @@ public ActionInvoker getActionInvoker(ProcessorDTO processor, BaseAction baseAct
         String endpoint = Optional.ofNullable(baseAction.getParameters().get(ENDPOINT_PARAM))
                 .orElseThrow(() -> buildNoEndpointException(processor));
         if (baseAction.getParameters().containsKey(USE_TECHNICAL_BEARER_TOKEN) && baseAction.getParameters().get(USE_TECHNICAL_BEARER_TOKEN).equals("true")) {
-            if (!webhookTechnicalBearerToken.isPresent()) {
-                throw new TechnicalBearerTokenNotConfiguredException("A webhook action needed the technical bearer token but it was not configured.");
+            Optional<AbstractOidcClient> abstractOidcClient = oidcClients.stream().filter(x -> x.getName().equals(OidcClientConstants.WEBHOOK_OIDC_CLIENT_NAME)).findFirst();
```
Comment (danielezonca) [inline actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java:61]: We should probably name the bean so that we can explicitly select which bean to inject. Probably current code is still necessary because `actions` are injected also in the manager but not used (we need to refactor this)

File: manager/src/test/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformerTest.java [hunk 2]
```diff
@@ -33,10 +33,10 @@
     private static final String TEST_CUSTOMER_ID = "test-customer";
     private static final String BRIDGE_ID = "br-01";
     private static final String BRIDGE_ENDPOINT = "http://www.example.com/bridge01";
-    private static final String BRIDGE_WEBHOOK = BRIDGE_ENDPOINT + "/events";
+    private static final String BRIDGE_WEBHOOK = BRIDGE_ENDPOINT;
     private static final String OTHER_BRIDGE_ID = "br-02";
     private static final String OTHER_BRIDGE_ENDPOINT = "http://www.example.com/bridge02";
-    private static final String OTHER_BRIDGE_WEBHOOK = OTHER_BRIDGE_ENDPOINT + "/events";
+    private static final String OTHER_BRIDGE_WEBHOOK = OTHER_BRIDGE_ENDPOINT;
```
Comment (danielezonca) [inline manager/src/test/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformerTest.java:39]: Why this change?

File: manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java [hunk 3]
```diff
@@ -42,8 +42,7 @@ public BaseAction transform(BaseAction action, String bridgeId, String customerI
         return transformedAction;
     }
 
-    static String getBridgeWebhookUrl(String bridgeEndpoint) throws MalformedURLException {
-        String fullUrl = String.join("", bridgeEndpoint, bridgeEndpoint.endsWith("/") ? "" : "/", "events");
```
Comment (danielezonca) [inline manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java:46]: Can you please clarify? It doesn't seem to be related with the PR

Comment (r00ta) [reply manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java:46]: It is not. From https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/502 the `/events` suffix is already present in the endpoint

Comment (r00ta) [reply manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java:46]: I mean, currently the sendToBridge action is broken on `main` because it sends events to `http://endpoint/events/events`. This is the fix and the tests needed to be changed accordingly

File: actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java (hunk 1 above)
Comment (r00ta) [reply actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java:61]: Yes, we need to refactor the `actions` module

File: manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java (hunk 3 above)
Comment (danielezonca) [reply manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java:46]: Ok thanks for the clarification 👍 

@jiripetrlik @sutaakar 
Do we have a ticket to implement a SendToBridge integration test?

File: actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java (hunk 1 above)
Comment (danielezonca) [reply actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java:61]: Can you please create a ticket? I think we should at refactor each action module to something like `action-sendtobridge-api`, `action-sendtobridge-validator` and `action-sendtobridge-runtime` (but maybe API and runtime are enough)

**Review by danielezonca — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java (hunk 3 above)
Comment (jomarko) [reply manager/src/main/java/com/redhat/service/bridge/manager/actions/sendtobridge/SendToBridgeActionTransformer.java:46]: @danielezonca we have
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/489
- https://issues.redhat.com/browse/MGDOBR-307

File: shard-operator/src/main/resources/application.properties [hunk 4]
```diff
@@ -18,8 +18,9 @@ event-bridge.sso.auth-server-url=${EVENT_BRIDGE_SSO_URL:http://${minikubeip}:300
 event-bridge.sso.client-id=${EVENT_BRIDGE_SSO_CLIENT_ID:shard-1}
 event-bridge.sso.credentials.secret=${EVENT_BRIDGE_SSO_SECRET:clhG7KwrwLk1rYr4yEjtjjiTM1iU8Vbc}
 
-event-bridge.webhook.technical-bearer-token=${WEBHOOK_TECHNICAL_BEARER_TOKEN:default}
-event-bridge.webhook.technical-account-id=${WEBHOOK_TECHNICAL_ACCOUNT_ID:402cf429-da04-4931-8089-e53ad452192b}
+event-bridge.webhook.client-id=${WEBHOOK_CLIENT_ID:webhook-1}
+event-bridge.webhook.client-secret=${WEBHOOK_CLIENT_SECRET:GMJiM0XNkp7b9bK0wP47odAtKxc4ufRv}
+event-bridge.webhook.account-id=${WEBHOOK_TECHNICAL_ACCOUNT_ID:402cf429-da04-4931-8089-e53ad452192b}
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/resources/application.properties:23]: these are the "secrets" of the public keycloak realm we use locally. They are not really secrets and are not used in any remote environment

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/webhook/site/WebhookSiteResource.java [hunk 5]
```diff
@@ -27,6 +27,6 @@ public static void deleteRequest(final WebhookSiteRequest request) {
                         ENDPOINT_UUID,
                         request.getUuid())
                 .then()
-                .statusCode(204);
+                .statusCode(200);
```
[the remark this answers is not in the document]
Comment (r00ta) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/webhook/site/WebhookSiteResource.java:30]: I need to rebase

[the remark this answers is not in the document]
Comment (r00ta) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/webhook/site/WebhookSiteResource.java]: gone

File: actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java [hunk 6]
```diff
@@ -56,10 +58,11 @@ public ActionInvoker getActionInvoker(ProcessorDTO processor, BaseAction baseAct
...
+            if (abstractOidcClient.isEmpty()) {
+                throw new TechnicalBearerTokenNotConfiguredException("A webhook action needed the webhook oidc client bean but it was not configured.");
             }
-            return new WebhookInvoker(endpoint, client, webhookTechnicalBearerToken.get());
+            return new WebhookInvoker(endpoint, client, abstractOidcClient.get());
```
[the remark this answers is not in the document]
Comment (r00ta) [reply actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java]: done
